crypto/thr_id.c

```diff
@@ -199,6 +204,8 @@ void CRYPTO_THREADID_current(CRYPTO_THREADID *id)
     /* Else pick a backup */
 #if defined(OPENSSL_SYS_WIN32)
     CRYPTO_THREADID_set_numeric(id, (unsigned long)GetCurrentThreadId());
+#elif defined(__unix__) || defined(unix)
+    CRYPTO_THREADID_set_numeric(id, (unsigned long)pthread_self());
```

Comment:
pthread_self() returns a value of type pthread_t.

My manual page for pthread_self() on this Ubuntu system has:

    NOTES
           POSIX.1 allows an implementation wide freedom in choosing  the  type  used  to  represent  a
           thread  ID;  for  example,  representation using either an arithmetic type or a structure is
           permitted.  Therefore, variables of type pthread_t can't portably be compared  using  the  C
           equality operator (==); use pthread_equal(3) instead.

           Thread identifiers should be considered opaque: any attempt to use a thread ID other than in
           pthreads calls is nonportable and can lead to unspecified results.

The usage in this pull request of casting the pthread_t to unsigned long could result in duplicate crypto thread ids depending on how the implementation chooses to represent pthread_t.